### ISSUE TITLE
Random Crashes on Linux

### ISSUE THREAD

**Issue by WuJune** [zed-industries/zed#12427]
### Check for existing issues

- [x] Completed

### Describe the bug / provide steps to reproduce it

Random Crashes on Linux after running for about 20 or 30 minute everty time, working with cpp files.
May have started from the `1716936798` pulled build version. I have built it multiple times in the past two days.

`~/.zsh_history`: 
```
: 1716904729:0;git pull
: 1716904900:0;cargo build --release

: 1716936798:0;git pull
: 1716936998:0;cargo build --release

: 1716983741:0;git pull
: 1716983772:0;cargo build --release
```

### Environment

OS: Fedora Linux 40 (Workstation Edition) x86_64
Host: TM1701
Kernel: Linux 6.8.10-300.fc40.x86_64
Uptime: 1 hour, 36 mins
Packages: 2461 (rpm)
Shell: zsh 5.9
Display (BOE0747): 1920x1080 @ 60Hz
DE: Gnome 46.1
WM: Mutter (X11)
WM Theme: Fluent-Light
Theme: Fluent-Light [GTK2/3/4]
Icons: Papirus [GTK2/3/4]
Font: Cantarell (11pt) [GTK2/3/4]
Cursor: Adwaita (24px)
Terminal: GNOME Terminal 3.50.1
Terminal Font: MesloLGMDZ Nerd Font (12pt)
CPU: Intel(R) Core(TM) i7-8550U (8) @ 4.00 GHz
GPU 1: Intel UHD Graphics 620 @ 1.15 GHz [Integrated]
GPU 2: NVIDIA GeForce MX150 [Discrete]
Memory: 5.60 GiB / 15.50 GiB (36%)
Swap: 267.25 MiB / 8.00 GiB (3%)
Disk (/): 92.72 GiB / 99.00 GiB (94%) - btrfs
Battery: 100% [AC Connected]
Locale: en_US.UTF-8

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

[2024-05-29T20:24:21+08:00 ERROR util] crates/theme/src/settings.rs:446: theme not found: Material Dark
[2024-05-29T20:24:22+08:00 ERROR util] crates/assistant/src/assistant_panel.rs:148: No such file or directory (os error 2)
[2024-05-29T20:24:22+08:00 ERROR util] crates/copilot/src/copilot.rs:1015: status error 403, response: "{\"message\":\"API rate limit exceeded for 38.150.12.151. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)\",\"documentation_url\":\"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting\"}\n"
[2024-05-29T20:24:22+08:00 ERROR auto_update] auto-update failed: error:error deserializing release

Caused by:
    expected value at line 1 column 1
[2024-05-29T20:34:16+08:00 ERROR util] crates/project/src/project.rs:3478: no worktree found for diagnostics path "/usr/include/fcntl.h"
[2024-05-29T20:45:44+08:00 ERROR workspace::item] unexpected item event after pane was dropped
[2024-05-29T20:45:44+08:00 ERROR workspace::item] unexpected item event after pane was dropped
[2024-05-29T20:45:44+08:00 ERROR workspace::item] unexpected item event after pane was dropped
[2024-05-29T20:45:44+08:00 ERROR workspace::item] unexpected item event after pane was dropped
Thread "main" panicked with "Unexpected descriptor allocation error: ERROR_FRAGMENTED_POOL" at /home/user_name/.cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/9c9cabf/blade-graphics/src/vulkan/descriptor.rs:85:27
   0: Zed::reliability::init_panic_hook::{closure#0}
             at Documents/GitHub/zed/crates/zed/src/reliability.rs:54:29
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2034:9
      std::panicking::rust_panic_with_hook
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:783:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:657:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:171:18
   4: rust_begin_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:645:5
   5: core::panicking::panic_fmt
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/panicking.rs:72:14
   6: <blade_graphics::hal::Device>::allocate_descriptor_set
             at .cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/9c9cabf/blade-graphics/src/vulkan/descriptor.rs:85:27
   7: <blade_graphics::hal::PipelineEncoder as blade_graphics::traits::PipelineEncoder>::bind::<gpui::platform::blade::blade_renderer::ShaderMonoSpritesData>
             at .cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/9c9cabf/blade-graphics/src/vulkan/command.rs:706:22
      <blade_graphics::hal::PipelineEncoder>::bind::<gpui::platform::blade::blade_renderer::ShaderMonoSpritesData>
             at .cargo/git/checkouts/blade-b2bcd1de1cf7ab6a/9c9cabf/blade-graphics/src/vulkan/command.rs:695:1
      <gpui::platform::blade::blade_renderer::BladeRenderer>::draw
             at Documents/GitHub/zed/crates/gpui/src/platform/blade/blade_renderer.rs:635:33
   8: <gpui::platform::linux::x11::window::X11Window as gpui::platform::PlatformWindow>::draw
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/window.rs:892:9
   9: <gpui::window::WindowContext>::present
             at Documents/GitHub/zed/crates/gpui/src/window.rs:1393:9
      <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}
             at Documents/GitHub/zed/crates/gpui/src/window.rs:695:33
      <gpui::app::AppContext as gpui::Context>::update_window::<(), <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}>::{closure#0}
             at Documents/GitHub/zed/crates/gpui/src/app.rs:1339:26
      <gpui::app::AppContext>::update::<core::result::Result<(), anyhow::Error>, <gpui::app::AppContext as gpui::Context>::update_window<(), <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}>::{closure#0}>
             at Documents/GitHub/zed/crates/gpui/src/app.rs:361:22
      <gpui::app::AppContext as gpui::Context>::update_window::<(), <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}>
             at Documents/GitHub/zed/crates/gpui/src/app.rs:1330:9
      <gpui::app::async_context::AsyncAppContext as gpui::Context>::update_window::<(), <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}>
             at Documents/GitHub/zed/crates/gpui/src/app/async_context.rs:91:14
      <gpui::window::AnyWindowHandle>::update::<gpui::app::async_context::AsyncAppContext, (), <gpui::window::Window>::new::{closure#3}::{closure#1}::{closure#0}>
             at Documents/GitHub/zed/crates/gpui/src/window.rs:4487:9
      <gpui::window::Window>::new::{closure#3}::{closure#1}
             at Documents/GitHub/zed/crates/gpui/src/window.rs:693:30
  10: util::measure::<(), <gpui::window::Window>::new::{closure#3}::{closure#1}>
             at Documents/GitHub/zed/crates/util/src/util.rs:325:9
      <gpui::window::Window>::new::{closure#3}
             at Documents/GitHub/zed/crates/gpui/src/window.rs:691:21
  11: <alloc::boxed::Box<dyn core::ops::function::FnMut<(), Output = ()>> as core::ops::function::FnMut<()>>::call_mut
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/alloc/src/boxed.rs:2027:9
      <gpui::platform::linux::x11::window::X11WindowStatePtr>::refresh
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/window.rs:539:13
      <gpui::platform::linux::x11::client::X11Client>::handle_event
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/client.rs:505:17
  12: <gpui::platform::linux::x11::client::X11Client>::xim_handle_event
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/client.rs:811:17
      <gpui::platform::linux::x11::client::X11Client>::new::{closure#5}
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/client.rs:334:33
      <core::cell::RefCell<calloop::sources::DispatcherInner<calloop::sources::generic::Generic<calloop::sources::generic::FdWrapper<alloc::rc::Rc<x11rb::xcb_ffi::XCBConnection>>, gpui::platform::linux::x11::client::EventHandlerError>, <gpui::platform::linux::x11::client::X11Client>::new::{closure#5}>> as calloop::sources::EventDispatcher<gpui::platform::linux::x11::client::X11Client>>::process_events::{closure#0}
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/sources/mod.rs:327:61
      <calloop::sources::generic::Generic<calloop::sources::generic::FdWrapper<alloc::rc::Rc<x11rb::xcb_ffi::XCBConnection>>, gpui::platform::linux::x11::client::EventHandlerError> as calloop::sources::EventSource>::process_events::<<core::cell::RefCell<calloop::sources::DispatcherInner<calloop::sources::generic::Generic<calloop::sources::generic::FdWrapper<alloc::rc::Rc<x11rb::xcb_ffi::XCBConnection>>, gpui::platform::linux::x11::client::EventHandlerError>, <gpui::platform::linux::x11::client::X11Client>::new::{closure#5}>> as calloop::sources::EventDispatcher<gpui::platform::linux::x11::client::X11Client>>::process_events::{closure#0}>
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/sources/generic.rs:290:9
      <core::cell::RefCell<calloop::sources::DispatcherInner<calloop::sources::generic::Generic<calloop::sources::generic::FdWrapper<alloc::rc::Rc<x11rb::xcb_ffi::XCBConnection>>, gpui::platform::linux::x11::client::EventHandlerError>, <gpui::platform::linux::x11::client::X11Client>::new::{closure#5}>> as calloop::sources::EventDispatcher<gpui::platform::linux::x11::client::X11Client>>::process_events
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/sources/mod.rs:326:9
  13: <calloop::loop_logic::EventLoop<gpui::platform::linux::x11::client::X11Client>>::dispatch_events
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/loop_logic.rs:445:31
      <calloop::loop_logic::EventLoop<gpui::platform::linux::x11::client::X11Client>>::dispatch::<core::option::Option<core::time::Duration>>
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/loop_logic.rs:559:9
      <calloop::loop_logic::EventLoop<gpui::platform::linux::x11::client::X11Client>>::run::<<gpui::platform::linux::x11::client::X11Client as gpui::platform::linux::platform::LinuxClient>::run::{closure#0}, core::option::Option<core::time::Duration>>
             at .cargo/registry/src/index.crates.io-6f17d22bba15001f/calloop-0.12.4/src/loop_logic.rs:596:13
      <gpui::platform::linux::x11::client::X11Client as gpui::platform::linux::platform::LinuxClient>::run
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/x11/client.rs:1076:9
  14: <gpui::platform::linux::x11::client::X11Client as gpui::platform::Platform>::run
             at Documents/GitHub/zed/crates/gpui/src/platform/linux/platform.rs:143:9
  15: <gpui::app::App>::run::<Zed::main::{closure#2}>
             at Documents/GitHub/zed/crates/gpui/src/app.rs:140:9
  16: Zed::main
             at Documents/GitHub/zed/crates/zed/src/main.rs:328:5
  17: <fn() as core::ops::function::FnOnce<()>>::call_once
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:250:5
      std::sys_common::backtrace::__rust_begin_short_backtrace::<fn(), ()>
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/sys_common/backtrace.rs:155:18
  18: std::rt::lang_start::<()>::{closure#0}
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/rt.rs:166:18
  19: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/core/src/ops/function.rs:284:13
      std::panicking::try::do_call
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
      std::panicking::try
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
      std::panic::catch_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
      std::rt::lang_start_internal::{{closure}}
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/rt.rs:148:48
      std::panicking::try::do_call
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:552:40
      std::panicking::try
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panicking.rs:516:19
      std::panic::catch_unwind
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/panic.rs:146:14
      std::rt::lang_start_internal
             at /rustc/9b00956e56009bab2aa15d7bff10916599e3d6d6/library/std/src/rt.rs:148:20
  20: main
  21: __libc_start_call_main
  22: __libc_start_main_impl
  23: _start

**Comment by CharlesChen0823** [comment]
@kvark

**Comment by apricotbucket28** [comment]
Should be fixed by https://github.com/zed-industries/zed/pull/12411

**Comment by WuJune** [comment]
> Should be fixed by #12411

Thanks, waiting for merge.

**Comment by mikayla-maki** [comment]
This should be fixed.
